src/core/Table/TablePaginator.tsx

```diff
@@ -0,0 +1,335 @@
+        style={{ transition: 'none' }}
+        onClick={() => onPageChange(index)}
+        aria-current={index === currentPage}
+        tabIndex={tabIndex}
+        size={size}
+      >
+        {index + 1}
+      </Button>
+    ),
+    [size, currentPage, focusedIndex, onPageChange],
+  );
+
+  const totalPagesCount = Math.ceil(totalRowsCount / pageSize);
+  const pageList = React.useMemo(
+    () =>
+      [...Array(totalPagesCount).fill(null)].map((_, index) =>
+        pageButton(index),
+      ),
+    [pageButton, totalPagesCount],
+  );
+  const [overflowRef, visibleCount] = useOverflow(pageList);
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    // alt + arrow keys are used by browser/assistive technologies
+    if (event.altKey) {
+      return;
+    }
+
+    const focusPage = (delta: number) => {
+      const newFocusedIndex = getBoundedValue(
+        focusedIndex + delta,
+        0,
+        totalPagesCount - 1,
+      );
+
+      if (focusActivationMode === 'auto') {
+        onPageChange(newFocusedIndex);
+      } else {
+        setFocusedIndex(newFocusedIndex);
+        const buttonToFocus = Array.from(
+          pageListRef.current?.querySelectorAll('.iui-button') ?? [],
+        ).find(
+          (el) => el.textContent?.trim() === (newFocusedIndex + 1).toString(),
+        );
+        (buttonToFocus as HTMLButtonElement | undefined)?.focus();
+      }
+    };
+
+    switch (event.key) {
+      case 'ArrowRight': {
+        focusPage(+1);
+        event.preventDefault();
+        break;
+      }
+      case 'ArrowLeft': {
+        focusPage(-1);
+        event.preventDefault();
+        break;
+      }
+      case 'Enter':
+      case ' ':
+      case 'Spacebar': {
+        event.preventDefault();
+        if (focusActivationMode === 'manual') {
+          onPageChange(focusedIndex);
+        }
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  const halfVisibleCount = Math.floor(visibleCount / 2);
+  let startPage = focusedIndex - halfVisibleCount;
+  let endPage = focusedIndex + halfVisibleCount + 1;
+  if (startPage < 0) {
+    endPage = Math.min(totalPagesCount, endPage + Math.abs(startPage));
+    startPage = 0;
+  }
+  if (endPage > totalPagesCount) {
+    startPage = Math.max(0, startPage - (endPage - totalPagesCount));
+    endPage = totalPagesCount;
+  }
+
+  const hasNoRows = totalPagesCount === 0;
+
+  const ellipsis = (
+    <span className={cx('iui-ellipsis', { 'iui-small': size === 'small' })}>
+      …
+    </span>
+  );
+
+  return (
+    <div className={cx('iui-paginator', className)} style={style} {...rest}>
+      <div className='iui-left' />
+      <div className='iui-center' style={{ minWidth: 0 }} ref={overflowRef}>
+        <IconButton
+          styleType='borderless'
+          disabled={currentPage === 0 || hasNoRows}
+          onClick={() => onPageChange(currentPage - 1)}
+          size={size}
+          aria-label='Previous page'
```

Comment:
Could add this to localization object maybe? Same with `Next page`

Reply:
Added.